incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -96,17 +100,79 @@ func (r *ObjectReconciler) GetGVK() schema.GroupVersionKind {
 	return r.GVK
 }
 
+// SetMode sets the Mode field of an object reconciler and syncs objects in the cluster if needed.
+// The method will return an error if syncs fail.
+func (r *ObjectReconciler) SetMode(ctx context.Context, mode api.SynchronizationMode, log logr.Logger) error {
+	log = log.WithValues("gvk", r.GVK)
+	newMode := r.getValidateMode(mode, log)
+	oldMode := r.Mode
+	if newMode == oldMode {
+		return nil
+	}
+	r.Log.V(1).Info("Changing Mode of the object reconciler", "old", oldMode, "new", newMode)
```

Comment:
This doesn't need to be V(1) since it will only occur rarely. In fact, this is probably a pretty important thing to log, especially if you get rid of the log message in the caller like I suggested.

Also, no need to capitalize "Mode."

Reply:
Thanks for the explanation. Done.